--- opencv/modules/highgui/src/window_QT.cpp	2018-06-30 01:04:51.649121800 +0300
+++ opencv/modules/highgui/src/window_QT_patched.cpp	2018-06-30 17:09:28.697256300 +0300
@@ -46,6 +46,30 @@
 
 #include <window_QT.h>
 
+#include <QAbstractEventDispatcher>
+#include <QFile>
+#include <QSizePolicy>
+#include <QInputDialog>
+#include <QBoxLayout>
+#include <QSettings>
+#include <QTimer>
+#include <QtConcurrentRun>
+#include <QWaitCondition>
+#include <QKeyEvent>
+#include <QMetaObject>
+#include <QLabel>
+#include <QIODevice>
+#include <QShortcut>
+#include <QStatusBar>
+#include <QFileInfo>
+#include <QDate>
+#include <QFileDialog>
+#include <QToolBar>
+
+#include <QAction>
+#include <QButtonGroup>
+#include <QMenu>
+
 #include <math.h>
 
 #ifdef _WIN32
@@ -60,7 +84,6 @@
     #endif
 #endif
 
-
 //Static and global first
 static GuiReceiver *guiMainThread = NULL;
 static int parameterSystemC = 1;
