diff --git a/CMakeLists.txt b/CMakeLists.txt
index af5e2c1..cee9726 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,6 +192,9 @@ if(MINGW)
   endif()
 endif()
 
+string(REPLACE "-arch i386" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+string(REPLACE "-arch i386" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+
 # ----------------------------------------------------------------------------
 # OpenCV cmake options
 # ----------------------------------------------------------------------------
